ScoutSuite/providers/azure/resources/resources.py

```diff
@@ -10,8 +10,7 @@ class AzureSimpleResources(SimpleResources, metaclass=abc.ABCMeta):
 
     async def fetch_all(self):
         raw_resource = await self.get_resources_from_api()
-        name, resource = self.parse_resource(raw_resource)
-        self[name] = resource
+        self.parse_resource(raw_resource)
```

Comment:
I don't want to be this guy, but I think I preferred the previous implementation. The reason is that `parse_resource` really sounds like it's going to return a parsed resource. It doesn't sound like it's going to sneakily add the resource to the dictionary. I think we should either rename the method, split the two responsibilities into two different methods, or fall back to the initial implementation.

One advantage of having a `parse_resource` returning a parsed resource and fetch_all handling the dictionary is that you reduce code duplication (you don't need to do `self.update` in every `parse_resource`).

Reply:
> It doesn't sound like it's going to sneakily add

That's one of my major griefs with the legacy code - a lot of methods have obscure names or are leveraged for operations that aren't immediately clear. bytes are cheap so don't be scared to use verbose names, and please add docstrings to clearly state what these classes/methods/functions are for.